### ISSUE TITLE
For LpsInstall

### ISSUE THREAD

**Issue by Eolien55** [NvChad/NvChad#458]
**Is your feature request related to a problem? Please describe.**
It's not related to a particular problem, but a feature that was removed (lspinstall plugin).

**Describe the solution you'd like**
Rather than just come back and cancelling the decision of removing the plugin, I'd like to see instructions to add it manually (or maybe, just putting the lines back and commenting these).

**Describe alternatives you've considered**
I know that the removal of that package was intended to un-bloat NvChad, and the creator may not want to change it.

**Additional context**
I tried myself to put the plugin in NvChad, but I've runned in some issues (that's why i suggest to give instructions for adding it).

**Screenshot**
None needed.

Anyway, even if you don't want to see this feature at all, even if optional, I want to say that this project's incredible. You're truly doing a great job, keep it up !

**Comment by A-Siam** [comment]
They removed the lspinstall method and replaced it with the alternative method in about 7hrs ago check this https://github.com/NvChad/site/commit/bae5633cdde2a52736798e66f75d14baaabe2d2d

The `config.md` referenced in the docs has a list of language servers and how to install them and then everything works as expected

**Comment by siduck76** [comment]
With lspconifg is easy too , trust me :/

**Comment by JoakoV3** [comment]
and what do I do after the npm command? still not getting lsp completitions.

**Comment by Eolien55** [comment]
> and what do I do after the npm command? still not getting lsp completitions.

Then (I guess), you go to `~/.config/nvim/lua/chadrc.lua` and at the very end of the file, there is the lines 
```M.plugins = {
     lspconfig = {
         servers ={}
     }
}``` ; there, you just add the server name between the curly braces, in quotes.

**Comment by dgmora** [comment]
I personally found it much easier to have LSPInstall and doing `LspInstall lua` instead of having to fiddle with downloading it and so on.

Maybe the documentation could be more clear? In [this page]https://nvchad.netlify.app/getting-started/post-install#install-lsp-servers) it may sounds as if adding the server name to `severs` gets you all set. Maybe it could make explicit that you still need to download the server and configure it with lspconfig

**Comment by NeckBeardPrince** [comment]
> I personally found it much easier to have LSPInstall and doing `LspInstall lua` instead of having to fiddle with downloading it and so on.
> 
> Maybe the documentation could be more clear? In [this page]https://nvchad.netlify.app/getting-started/post-install#install-lsp-servers) it may sounds as if adding the server name to `severs` gets you all set. Maybe it could make explicit that you still need to download the server and configure it with lspconfig

Same, here because nvim always knew where to execute the language server binary from. Now I have to add the directory for `yamlls` and `bashls` into my $PATH (so far I've only used those).

**Comment by abalmos** [comment]
I agree. This change appears to be trading so-called "bloat" for user effort. In my opinion, is a poor trade off. LSPs are so critical to a decent IDE experience, I personally struggle to see how this is off-topic for nvchad.

I use several languages, and I now have to manually update each LSP tool--which are often not packaged at the distro level or are too old. Everyone one of them has their own means of installing/update/building and not great means of knowing an update is available. It's like they assume people are using tools / IDE plugins to manage them :)

**Comment by siduck76** [comment]
> I agree. This change appears to be trading so-called "bloat" for user effort. In my opinion, is a poor trade off. LSPs are so critical to a decent IDE experience, I personally struggle to see how this is off-topic for nvchad.
> 
> I use several languages, and I now have to manually update each LSP tool--which are often not packaged at the distro level or are too old. Everyone one of them has their own means of installing/update/building and not great means of knowing an update is available. It's like they assume people are using tools / IDE plugins to manage them :)

we were getting multiple issues with lspinstall for windows users , Im aware there's another lspinstall plugin that supports windows too but yea my exams are going on so dont have the time to add it! Also the custom hooks PR is gonna merge in few days so the user will be able to integrate his config without getting them overwritten by updates , NvChad's meant to be a base config. managing lspservers manually isnt that hard imo .

**Comment by abalmos** [comment]
@siduck76 I think people are just questioning the logic behind removing it and giving feedback.

Sure, Windows folks can't use it and have had to use the current manual method, but Linux users could and it improved their workflows. Seems like a simple addition to the docs warning Windows users would have been easier on your time and left (likely the majority) the other users happy.

> Also the custom hooks PR is gonna merge in few days so the user will be able to integrate his config without getting them overwritten by updates

Great! I truly hope it makes tracking upstream a bit easier. I have already added `lspinstall` back-in (reverted the removal commit). 

I'm not trying to tell you how to manage the project, just giving you some feedback from the user side. I'm very glad you all made NvChad, overnight I went from a nasty vimrc hacked over many years to neovim with many of the features I knew were possible but didn't have the time to figure out.

>  NvChad's meant to be a base config.

I do appreciate that. Just seems to me that making LSP really easy ought to be a primary goal of a base config. Much more important than a few less milliseconds of load time IMHO.

> managing lspservers manually isnt that hard imo .

Is not about being _hard_, it's about being a waste of time. Previously, a simple keymapping would update all of the LSP servers. I could just do it once per day and know I'm running the latest. With languages like `rust` this is important. `rust_analyzer` improves with the hour somedays and releases at least weekly.

**Comment by siduck76** [comment]
If you use distros like arch and void then you can find many lsp servers as packages and install them

**Comment by Eolien55** [comment]
> If you use distros like arch and void then you can find many lsp servers as packages and install them 

Yes, but it is penalizing non-arch users, like the feature penalized windows users

**Comment by siduck76** [comment]
> > If you use distros like arch and void then you can find many lsp servers as packages and install them
> 
> Yes, but it is penalizing non-arch users, like the feature penalized windows users

agreed , maybe we can try lspinstaller sometime in the future . I had removed lspinstall because windows users were having issues with it and lsp stuff is a needed thing so wouldnt want windows users just change the config manually.

**Comment by dgmora** [comment]
But by removing it, now it's  users from every platform that needs to find a way to install the servers and modify the configuration, no?

For me as a vim user a big motivation to use something different than my own configuration is that configuring lsps is a pain, I'd love if it would come with even more support and already configured for every language (there are some similar projects that do so).

Anyway no big deal, we can also install LspInstall meanwhile, I still really like the project!

**Comment by JoakoV3** [comment]
> > > If you use distros like arch and void then you can find many lsp servers as packages and install them
> > 
> > 
> > Yes, but it is penalizing non-arch users, like the feature penalized windows users
> 
> agreed , maybe we can try lspinstaller sometime in the future . I had removed lspinstall because windows users were having issues with it and lsp stuff is a needed thing so wouldnt want windows users just change the config manually.

But then linux and mac users will just manually modify the configs to have that extension.

Maybe you can solve this by adding a new feature which gets the current os in lua script. Then if it is on windows comment the plugin and if it isn't just leave it. 

There's a stack overflow post about getting the current os with lua here: [link to overflow](https://stackoverflow.com/questions/295052/how-can-i-determine-the-os-of-the-system-from-within-a-lua-script)
